### PR TITLE
[MSHARED-944] Drop support for Maven 3.0. Require 3.1.1

### PR DESCRIPTION
**Pull request by slachiewicz**

### CODE REVIEW

**Comment by michael-o** [comment]
Massive, will need some time for this.

**Comment by slachiewicz** [comment]
there is nothing special - lots of removal of old maven 3.0 stuff + replacement of reflections to just us normal direct call to exposed API. No need to spent too much time here. 
CI is green https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-artifact-transfer/job/MSHARED-944/

**Comment by slachiewicz** [comment]
With these changes, it should be easier to move forward with maven-dependency-tree and other plugins - to not fight with old dependencies.

File: pom.xml [hunk 1]
```diff
@@ -165,12 +165,12 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
-      <version>3.0</version>
+      <version>3.1.1</version>
```
Comment (michael-o) [inline pom.xml:168]: I think this should be moved to a property.

**Review by elharo — APPROVED** [review]

**Review by michael-o — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (slachiewicz) [reply pom.xml:168]: Sure, will fix this next time

**Comment by rfscholte** [comment]
Please don't do this. The intention of maven-artifact-transfer is a bridge to any Aether implementation.

**Comment by rfscholte** [comment]
Sorry @slachiewicz if you've spend a lot of time on this, but I need to close this. Once everything is using Maven 3.1+ we can simply archive maven-artifact-transfer because it has become an unnecessary abstraction layer.

**Comment by slachiewicz** [comment]
Do we have anywhere documented this recommendation? I think we still have on wiki to use this artefact. 
Also with this changes we can easily switch to new aether impl  by dep update plus remove redundant code/simplify other projects.

**Comment by rfscholte** [comment]
I still consider adding the `org.apache.maven.shared.transfer.project` package a bad choice, because it introduces a dependency on Maven. 
https://cwiki.apache.org/confluence/display/MAVEN/Plugin+migration+to+Maven3+dependencies does indeed talk about adding maven-artifact-transfer, but https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=155749857 should contain the removal of it.

**Comment by olamy** [comment]
was a good idea to get RID of old dependencies until we go away correctly from org.eclipse package.

**Comment by cstamas** [comment]
Just FTR, now that https://github.com/apache/maven-artifact-transfer/pull/24 has been merged, and MAT modularized, same effect can be now achieved by simply excluding the 3.0 module from dependencies.

But, my problem is not (only) this: MAT is half done (is unfinished) and if we stick with it, there will be probably someone who will try to "improve" its API, thus, we are again at doubling efforts as new Maven 4 API and MAT will become somewhat "competing" APIs.

This is why I think just detaching from MAT and leave it to oblivion is better, as for Maven 3.1-3.999 using eclipse resolver packages are completely okay, while in Maven 4 we will need to switch to new API anyway....

**Pull request closed without merging**